core/include/cubos/core/ecs/system.hpp

```diff
@@ -77,108 +84,89 @@ namespace cubos::core::ecs
         template <typename R> struct SystemFetcher<R&>
         {
             using Type = WriteResource<R>;
+            using State = std::monostate;
 
-            /// Adds the resource to the system info structure.
-            /// @param info The system info structure to add to.
             static void add(SystemInfo& info);
-
-            /// @param world The world to fetch from.
-            /// @param commands The commands object.
-            /// @returns The requested resource write lock.
-            static WriteResource<R> fetch(World& world, CommandBuffer& commands);
-
-            /// @param lock The resource lock to get the reference from.
-            /// @returns The requested resource reference.
+            static State prepare(World& world);
+            static WriteResource<R> fetch(World& world, CommandBuffer& commands, State& state);
```

Comment:
Use the `Type` alias set previously on all applicable cases
```suggestion
            static Type fetch(World& world, CommandBuffer& commands, State& state);
```